gematria/granite/graph_builder_model_inference.cc

```diff
@@ -46,20 +50,88 @@ namespace {
 
 using ::tflite::FlatBufferModel;
 
-// The indices of the input tensors in the compiled TensorFlow Lite model. This
-// order of the input tensors must be preserved during the conversion of the
-// model to the .tflite format.
-constexpr int kDeltaBlockIndexTensor = 0;
-constexpr int kGraphNodesTensor = 1;
-constexpr int kGraphEdgesTensor = 2;
-constexpr int kGraphGlobalsTensor = 3;
-constexpr int kGraphReceiversTensor = 4;
-constexpr int kGraphSendersTensor = 5;
-constexpr int kGraphNEdgeTensor = 6;
-constexpr int kGraphNNodeTensor = 7;
-constexpr int kInstructionNodeMaskTensor = 8;
-
-constexpr int kNumInputTensors = 9;
+// Graph builder models may have some valid combination of these input tensors
+// depending on their configuration.
+enum class InputTensor {
+  kDeltaBlockIndexTensor = 0,
+  kGraphNodesTensor,
+  kGraphEdgesTensor,
+  kGraphGlobalsTensor,
+  kGraphReceiversTensor,
+  kGraphSendersTensor,
+  kGraphNEdgeTensor,
+  kGraphNNodeTensor,
+  kInstructionNodeMaskTensor,
```

Comment:
Optional: Since this is an enum class (and thus always requires the `InputTensor::` qualification), I think we can remove the `Tensor` at the end of the value names, and make them a bit shorter.